src/http.rs

```diff
@@ -42,61 +41,56 @@ async fn run(
-            .map(|query: Arc<Query>| {
-                let wallets = query.get_hd_wallets();
-                reply::json(&wallets)
-            });
+    let hd_wallets_handler = warp::get()
+        .and(warp::path!("wallet"))
+        .and(query.clone())
+        .map(|query: Arc<Query>| {
+            let wallets = query.get_hd_wallets();
+            reply::json(&wallets)
+        });
 
     // GET /hd/:fingerprint
     let hd_wallet_handler = warp::get()
-        .and(warp::path!("hd" / Fingerprint))
+        .and(warp::path!("wallet" / DescrChecksum))
         .and(query.clone())
-        .map(|fingerprint: Fingerprint, query: Arc<Query>| {
-            let wallet = query
-                .get_hd_wallet(fingerprint)
-                .or_err(StatusCode::NOT_FOUND)?;
+        .map(|descr_cs: DescrChecksum, query: Arc<Query>| {
+            let wallet = query.get_hd_wallet(descr_cs).or_err(StatusCode::NOT_FOUND)?;
             Ok(reply::json(&wallet))
         })
         .map(handle_error);
 
     // GET /hd/:fingerprint/:index
     let hd_key_handler = warp::get()
-        .and(warp::path!("hd" / Fingerprint / u32))
+        .and(warp::path!("wallet" / DescrChecksum / u32))
         .and(query.clone())
-        .map(|fingerprint: Fingerprint, index: u32, query: Arc<Query>| {
+        .map(|descr_cs: DescrChecksum, index: u32, query: Arc<Query>| {
             let script_info = query
-                .get_hd_script_info(fingerprint, index)
+                .get_hd_script_info(descr_cs, index)?
                 .or_err(StatusCode::NOT_FOUND)?;
             Ok(reply::json(&script_info))
         })
         .map(handle_error);
 
     // GET /hd/:fingerprint/gap
     let hd_gap_handler = warp::get()
-        .and(warp::path!("hd" / Fingerprint / "gap"))
+        .and(warp::path!("wallet" / DescrChecksum / "gap"))
         .and(query.clone())
-        .map(|fingerprint: Fingerprint, query: Arc<Query>| {
-            let gap = query
-                .find_hd_gap(fingerprint)
-                .or_err(StatusCode::NOT_FOUND)?;
+        .map(|descr_cs: DescrChecksum, query: Arc<Query>| {
+            let gap = query.find_hd_gap(descr_cs)?.or_err(StatusCode::NOT_FOUND)?;
             Ok(reply::json(&gap))
         })
         .map(handle_error);
 
     // GET /hd/:fingerprint/next
     let hd_next_handler = warp::get()
-        .and(warp::path!("hd" / Fingerprint / "next"))
+        .and(warp::path!("wallet" / DescrChecksum / "next"))
         .and(query.clone())
-        .map(|fingerprint: Fingerprint, query: Arc<Query>| {
+        .map(|descr_cs: DescrChecksum, query: Arc<Query>| {
             let wallet = query
-                .get_hd_wallet(fingerprint)
+                .get_hd_wallet(descr_cs.clone())
                 .or_err(StatusCode::NOT_FOUND)?;
             let next_index = wallet.get_next_index();
-            let uri = format!("/hd/{}/{}", fingerprint, next_index);
+            let uri = format!("/hd/{}/{}", descr_cs, next_index);
```

Comment:
`/hd/` should be changed too (and possibly some other places that construct a uri)

Reply:
Fixed here and in some comments.